File: DeepLearningServer/Classes/ToolStatusManager.cs
namespace DeepLearningServer.Classes
{
    public class ToolStatusManager
    {
        private static readonly string statusFilePath = "tool_status.txt";

        public static bool IsProcessRunning()
        {
            if (File.Exists(statusFilePath))
            {
                string content = File.ReadAllText(statusFilePath);
                return content.Trim() == "Running";
            }
            return false;
        }

        public static void SetProcessRunning(bool isRunning)
        {
            File.WriteAllText(statusFilePath, isRunning ? "Running" : "Idle");
        }

    }
}

File: deeplearningserver-python/services/tool_status_manager.py
from pathlib import Path


class ToolStatusManager:
    _status_file = Path("tool_status.txt")

    @classmethod
    def is_process_running(cls) -> bool:
        if not cls._status_file.exists():
            return False
        return cls._status_file.read_text(encoding="utf-8").strip() == "Running"

    @classmethod
    def set_process_running(cls, is_running: bool) -> None:
        cls._status_file.write_text("Running" if is_running else "Idle", encoding="utf-8")
